File: docs/guides/python/saveObjectsPublicUser.py
from algoliasearch.search.client import SearchClientSync
from json import loads


playlists = []  # Your records

_client = SearchClientSync("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")


_client.save_objects(
    index_name="<YOUR_INDEX_NAME>",
    objects=playlists,
    wait_for_tasks=False,
    batch_size=1000,
    request_options={
        "headers": loads("""{"X-Algolia-User-ID":"*"}"""),
    },
)

File: docs/guides/dart/lib/saveObjectsPublicUser.dart
import 'package:algolia_client_search/algolia_client_search.dart';

final playlists = [/* Your records */];

void saveObjectsPublicUser() async {
  final client =
      SearchClient(appId: 'ALGOLIA_APPLICATION_ID', apiKey: 'ALGOLIA_API_KEY');

  final batchParams = BatchWriteParams(
      requests: playlists
          .map((record) => BatchRequest(
                action: Action.addObject,
                body: record,
              ))
          .toList());
  await client.batch(
      indexName: "<YOUR_INDEX_NAME>",
      batchWriteParams: batchParams,
      requestOptions: RequestOptions(
        headers: {
          'X-Algolia-User-ID': "*",
        },
      ));
}
